feat: additional work sections

sashokbg opened the pull request:
I would like to open this PR as a discussion on how to better arrange work experiences along multiple pages.

I have added a button "Add new section" under the current "work" section that adds a work-1, work-2 and so on sections.
I have modified the WorkModal to work with changing path (similar to the custom sections).

Please let me know what you think.

https://gfycat.com/thesesecretamericanratsnake

AmruthPillai commented:
This is an interesting idea, and thank you very much for the screen cap demonstrating your feature implementation. That saves a lot of time.

I am just curious about the naming. "Add New Section" is a bit ambiguous as there's already a "Add New Section" in the footer. Would "Duplicate Section" be more appropriate? (this also has the confusion that items in the section would be copied over). Any other suggestions come to mind?

Once again, thank you so much for your contributions to Reactive Resume <3

sashokbg commented:
Hello @AmruthPillai,

Yes you are right I will change the message.
I was thinking that any section should in theory be duplicate-able.

We should also make sure that users can clearly understand that duplicating a section is so that it can be arranged in the layout on different pages. Maybe a on-hover tooltip ? What do you think ?

Best Regards and thank you for the motivating message !
Alex

sashokbg commented:
Hello @AmruthPillai I have cleaned the code a bit and rendered it more generic. I also added the label "Duplicate Section" in English only so far.
Please let me know what you think.

sashokbg commented on client/components/build/LeftSidebar/sections/Section.tsx:
```diff
@@ -10,28 +10,34 @@ import Heading from '@/components/shared/Heading';
 
 type Props = {
   path: `sections.${string}`;
+  type?: SectionType;
```
I needed to add type to sections so that we can have multiple work sections that share the same modal by type but have different paths

sashokbg commented on client/components/build/LeftSidebar/sections/Section.tsx:
```diff
@@ -42,21 +48,42 @@ const Section: React.FC<Props> = ({
+    let modal: ModalName = validate(id) ? 'builder.sections.custom' : `builder.${path}`;
 
+    if (type) {
+      modal = `builder.sections.${type}`;
```
Find modal by type instead of path. This way we can have one modal "**builder.sections.work**" that works with paths "**builder.sections.work-1, work-2**" and so on

AmruthPillai commented:
@sashokbg I'll have a look over this PR and the other one on the weekend as I'm tied up with work right now. Just letting you know if you're wondering what's taking so long :D But thank you so much for your amazing contribution to the project. It really means a lot, to me and the community.
